theseus/src/api/jre.rs

```diff
@@ -0,0 +1,136 @@
+//! Authentication flow interface
+use crate::{
+    prelude::Profile,
+    state::JavaGlobals,
+    util::jre::{self, extract_java_majorminor_version, JavaVersion},
+    State,
+};
+use daedalus as d;
+
+pub const JAVA_8_KEY: &str = "JAVA_8";
+pub const JAVA_17_KEY: &str = "JAVA_17";
+pub const JAVA_18PLUS_KEY: &str = "JAVA_18PLUS";
+
+// Autodetect JavaSettings default
+// Make a guess for what the default Java global settings should be
+pub fn autodetect_java_globals() -> crate::Result<JavaGlobals> {
+    let mut java_8 = find_java8_jres()?;
+    let mut java_17 = find_java17_jres()?;
+    let mut java_18plus = find_java18plus_jres()?;
+
+    // Simply select last one found for initial guess
+    let mut java_globals = JavaGlobals::new();
+    if let Some(jre) = java_8.pop() {
+        java_globals.insert(JAVA_8_KEY.to_string(), jre);
+    }
+    if let Some(jre) = java_17.pop() {
+        java_globals.insert(JAVA_17_KEY.to_string(), jre);
+    }
+    if let Some(jre) = java_18plus.pop() {
+        java_globals.insert(JAVA_18PLUS_KEY.to_string(), jre);
+    }
+
+    Ok(java_globals)
+}
+
+// Gets the optimal JRE key for the given profile, using Daedalus
+// Generally this would be used for profile_create, to get the optimal JRE key
+// this can be overwritten by the user a profile-by-profile basis
+pub async fn get_optimal_jre_key(profile: &Profile) -> crate::Result<String> {
+    let state = State::get().await?;
+
+    // Fetch version info from stored profile game_version
+    let version = state
+        .metadata
+        .minecraft
+        .versions
+        .iter()
+        .find(|it| it.id == profile.metadata.game_version.as_ref())
+        .ok_or_else(|| {
+            crate::ErrorKind::LauncherError(format!(
+                "Invalid or unknown Minecraft version: {}",
+                profile.metadata.game_version
+            ))
+        })?;
+
+    // Get detailed manifest info from Daedalus
+    let version_info = d::minecraft::fetch_version_info(version).await?;
```

Comment:
this is cached in meta/versions. iirc there is a function in the launcher module which will fetch the cached version if needed